src/crypto/crypto_ec.cc

```diff
@@ -220,17 +222,23 @@ void ECDH::ComputeSecret(const FunctionCallbackInfo<Value>& args) {
     return;
   }
 
-  // NOTE: field_size is in bits
-  int field_size = EC_GROUP_get_degree(ecdh->group_);
-  size_t out_len = (field_size + 7) / 8;
-  AllocatedBuffer out = AllocatedBuffer::AllocateManaged(env, out_len);
+  std::unique_ptr<BackingStore> bs;
+  {
+    NoArrayBufferZeroFillScope no_zero_fill_scope(env->isolate_data());
+    // NOTE: field_size is in bits
+    int field_size = EC_GROUP_get_degree(ecdh->group_);
+    size_t out_len = (field_size + 7) / 8;
+    bs = ArrayBuffer::NewBackingStore(env->isolate(), out_len);
+  }
 
-  int r = ECDH_compute_key(
-      out.data(), out_len, pub.get(), ecdh->key_.get(), nullptr);
-  if (!r)
+  if (!ECDH_compute_key(
+          bs->Data(), bs->ByteLength(), pub.get(), ecdh->key_.get(), nullptr))
     return THROW_ERR_CRYPTO_OPERATION_FAILED(env, "Failed to compute ECDH key");
 
-  args.GetReturnValue().Set(out.ToBuffer().FromMaybe(Local<Value>()));
+  Local<ArrayBuffer> ab = ArrayBuffer::New(env->isolate(), std::move(bs));
+  Local<Value> buffer;
+  if (!Buffer::New(env, ab, 0, ab->ByteLength()).ToLocal(&buffer)) return;
+  args.GetReturnValue().Set(buffer);
```

Comment:
nit: why not if the other way around and have one less (possibly hidden) `return`?
```suggestion
  if (Buffer::New(env, ab, 0, ab->ByteLength()).ToLocal(&buffer)) {
   args.GetReturnValue().Set(buffer);
  }
```

Reply:
@dygabo this is using the "early return" pattern where we return from a function as early as possible in case of an error (a `false` return value from `ToLocal()` indicates that an exception has been scheduled to be thrown from V8 and invoking most of the V8 APIs at this point would lead to crashes, so we should return control to V8 asap, so that the exception can be thrown at the JS level).